Config.py

```diff
@@ -0,0 +1,242 @@
+import os
+import time
+import urllib2
+from ConfigParser import SafeConfigParser, NoOptionError, NoSectionError
+import logging
+from logging.handlers import RotatingFileHandler, MemoryHandler, SMTPHandler
+
+from Errors import TLDsListUnavailableError, InsufficientPermissionError
+
+class STARTTLSEverywhereConfig(SafeConfigParser):
+  default_cfg_path = "/etc/starttls-everywhere/starttls-everywhere.cfg"
+
+  custom_defaults = {
+    "postfix": {
+      "cfg_dir": "/etc/postfix",
+      "ca_path": "%(capath)s",
+      "main_config_file": "main.cf",
+      "policy_defs_file": "starttls-everywhere",
+      "smtp_tls_policy_maps_type": "btree",
+      "postmap_path": "/usr/sbin/postmap",
+      "tafile_real_dir": "/var/spool/postfix/etc",
+      "tafile_dir": "/etc"
+    }
+  }
+
+  def __init__(self):
+    self.logger = None
+
+    SafeConfigParser.__init__(self, {
+      # general
+
+      "data_dir": "/var/lib/starttls-everywhere",
+      "guessstarttlspolicy_cache_dir": "%(data_dir)s/guessstarttlspolicy_cache",
+      "openssl_path": "openssl",
+      "capath": "/etc/ssl/certs/",
+      "tlds_update_interval": "86400",
+      "tlds_url": "https://data.iana.org/TLD/tlds-alpha-by-domain.txt",
+      "diff_cmd": "diff -y {new} {old}",
+      "failure_threshold_percent": "0.1",
+      "log_level": "INFO",
+      "log_file_level": "%(log_level)s",
+      "log_smtp_level": "ERROR",
+      "logwatcher_reports_dir": "%(data_dir)s/logwatcher-reports",
+      "logwatcher_reports_fmt": "%%Y-%%m-%%d-%%H-%%M-%%S.log",
+      })
+
+  def get(self,section,option,default=None):
+    try:
+      return SafeConfigParser.get(self,section,option)
+    except (NoSectionError, NoOptionError) as e:
+      if default is not None:
+        return default
+
+      if section in STARTTLSEverywhereConfig.custom_defaults:
+        if isinstance(e,NoSectionError):
+          self.add_section(section)
+
+        if option in STARTTLSEverywhereConfig.custom_defaults[section]:
+          return SafeConfigParser.get(self,section,option,vars={
+            option: STARTTLSEverywhereConfig.custom_defaults[section][option]})
+      raise
+    except:
+      raise
+
+  def get_tlds_list(self):
```

Comment:
I really like the idea of doing this verification, currently validation is all performed in the new Config container and this pulled out into a function thereabouts could be used *somewhere* to add extra sanity checking to domains in the config.